enumerable_methods.rb

```diff
@@ -0,0 +1,115 @@
+# rubocop:disable all
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    item = self if instance_of?(Array)
+    item = to_a if instance_of?(Range) || Hash
+
+    i = 0
+    while i < item.length
+      yield(item[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    item = to_a
+    i = 0
+    while i < var.length
+      yield(var[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    array = []
+    my_each { |item| array << item if yield item }
+    array
+  end
+
+  def my_all?(argument = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif argument.nil?
+      my_each { |item| return false if item.nil? || item == false }
+    elsif !argument.nil? && (argument.is_a? Class)
+      my_each { |item| return false unless [item.class, item.class.superclass].include?(argument) }
+    elsif argument.class && !argument.nil == Regexp
+      my_each { |item| return false unless argument.match(item) }
+    else 
+      my_each { |item| return false if item != argument }
+    end
+    true
+  end
+
+  def my_any?(argument = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      return false
+    elsif argument.nil?
+      my_each { |item| return true if item }
+    elsif !argument.nil? && argument.is_a?(Class)
+      my_each { |item| return true unless [item.class, item.class.superclass].include?(argument) }
+    elsif argument.class && !argument.nil == Regexp
```

Comment:
It's `nil?` not `nil`

Even after fixing that bugs, `#my_any?` is not working like expected.